### ISSUE TITLE
v2 regression: delegate_to magic variable no longer exists

### ISSUE THREAD

**Issue by nikut** [ansible/ansible#15546]
##### ISSUE TYPE
- Bug Report
##### ANSIBLE VERSION

```
ansible 2.0.2.0 (detached HEAD 24d9e5e0b4) last updated 2016/04/20 10:08:39 (GMT +300)
  lib/ansible/modules/core: (detached HEAD e1cedb1633) last updated 2016/04/20 10:08:50 (GMT +300)
  lib/ansible/modules/extras: (detached HEAD 4eb177e545) last updated 2016/04/20 10:08:51 (GMT +300)
  config file = /etc/ansible/ansible.cfg
```
##### CONFIGURATION

N/A
##### OS / ENVIRONMENT

N/A
##### SUMMARY

The `delegate_to` magic variable documented in http://docs.ansible.com/ansible/playbooks_variables.html#magic-variables-and-how-to-access-information-about-other-hosts has disappeared in Ansible v2.
##### STEPS TO REPRODUCE

Playbook:

```
- hosts: testhost
  gather_facts: no
  tasks:
    - debug: var=delegate_to
      delegate_to: localhost

```

##### EXPECTED RESULTS

Ansible 1.9.6:

```
TASK: [debug var=delegate_to] *************************************************
ok: [testhost -> localhost] => {
    "var": {
        "delegate_to": "localhost"
    }
}

```
##### ACTUAL RESULTS

Ansible 2.0.2:

```
TASK [debug] *******************************************************************
ok: [testhost -> localhost] => {
    "delegate_to": "VARIABLE IS NOT DEFINED!"
}

```

**Comment by bcoca** [comment]
# Possible Misunderstanding

Hi!

Thanks very much for your submission to Ansible.  It sincerely means a lot to us.

We believe the ticket you have filed is being somewhat misunderstood, as one thing works a little differently than stated.

This was not a feature, in previous versions directives 'leaked' into the vars, we fixed that so this won't show anymore.  Use `ansible_host` if you want to reference the host we delegate_to.

In the future, this might be a topic more well suited for the user list, which you can also post here if you'd like some more help with the above.
- https://groups.google.com/forum/#!forum/ansible-project

Thank you once again for this and your interest in Ansible!

**Comment by nikut** [comment]
Thanks for pointing out `ansible_host`. 

If this is not a bug, it is at least a breaking change in functionality. It's a documented feature that used to work in 1.x but no longer does in 2.x so playbooks that relied on it will break. `ansible_host` does not exist in 1.x so there is no common way to achieve this in both 1.x and 2.x, you have to employ a workaround such as `ansible_host | default(delegate_to)`. I guess that's fine but it should at least be documented. 

Maybe I'll submit a docs PR later.

**Comment by bcoca** [comment]
documented? can you show where?

To do a version that would work across versions use `ansible_ssh_host`

**Comment by nikut** [comment]
The link to the documentation is in my bug report above. IIRC that particular sentence may in fact have been added by me after being asked to do so on IRC by you guys. :)

**Comment by bcoca** [comment]
We might have been wrong when making that request, 'delegate_to' was never intentionally made a special var as ansible_ssh_host/ansible_host should actually reflect that info and are intentional.

Sadly before 2.x most directives 'bled' into the vars which caused many issues, in 2.x we fixed this and made them separate, so it was considered a bug, though you might have relied on it as a feature.

I'll leave this open and bring up as a discussion topic with the rest of the core team.

**Comment by nikut** [comment]
Thanks for the clarification. I'm fine with ansible_(ssh_)host so I guess this essentially boils down to a documentation fix to remove delegate_to from magic variables and maybe mention somewhere that ansible_(ssh_)host can be used to find out where a task was delegated to.

**Comment by bcoca** [comment]
closing via merged #15546
